scalar/scalar.go

```diff
@@ -0,0 +1,30 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package scalar
+
+// Scalar represent a set of useful types to be used with type parameters.
+//
+// For now we only support the type that are frequently used within the
+// controller repositories, code-generator and runtime.
+type Scalar interface {
+	int | int32 | int64 | float32 | float64 | bool | string
+}
```

Comment:
Having seconds thoughts, maybe this package should only contain the `Scalar` constraint, and we should have `compare`/`strutil` import it in order to extend the functions defined within them to be applicable beyond `string` type only